core/shared/src/main/scala/zio/prelude/Ord.scala

```diff
@@ -961,17 +878,76 @@ sealed trait Ordering { self =>
     }
 
   /**
-   * Converts this `Ordering` to an ordinal representation, with `0`
-   * representing `LessThan`, `1` representing `Equals` and `2` representing
-   * `GreaterThan`.
+   * Returns this ordering, but if this ordering is equal returns the
+   * specified ordering.
    */
-  final def ordinal: Int =
+  def orElse(that: => PartialOrdering): PartialOrdering =
     self match {
-      case Ordering.LessThan    => 0
-      case Ordering.Equals      => 1
-      case Ordering.GreaterThan => 2
+      case Ordering.Equals => that
+      case ordering        => ordering
     }
 
+  def reduce(that: PartialOrdering): PartialOrdering = (self, that) match {
+    case (Ordering.LessThan, Ordering.LessThan)       => Ordering.LessThan
+    case (Ordering.GreaterThan, Ordering.GreaterThan) => Ordering.GreaterThan
+    case (Ordering.Equals, that)                      => that
+    case (self, Ordering.Equals)                      => self
+    case _                                            => PartialOrdering.NoOrder
+  }
+}
+
+sealed trait Comparison extends Product with Serializable
+
+object Comparison {
+
+  sealed trait NotEqual extends Comparison
+
+}
+
+object PartialOrdering {
```

Comment:
This "overlapping" ADT is interesting but I wonder what problems it will cause with type inference. In particular, there are some cases where we would like a partial ordering, but due to specific return types, we will get an ordering.

I understand it is convenient to do it like this and I don't want to minimize that. But we should be open to changing our mind one way or the other based on tests.

Reply:
Another alternative would be to have a separate method for a `PartialOrd`:

```scala
trait PartialOrd[A] {
  def compareOption(l: A, r: R): Option[Ordering]
}
```

Then we wouldn't need a separate ADT and we wouldn't have any overlaps.

@sideeffffect What do you think?